src/Microsoft.SqlTools.ServiceLayer/LanguageExtensibility/ExternalLanguageOperations.cs

```diff
@@ -0,0 +1,341 @@
+        private const string GetAllScript = @"
+SELECT l.external_language_id, language, l.create_date, dp.name, content, file_name, platform_desc, parameters, environment_variables
+FROM sys.external_languages l 
+JOIN sys.external_language_files lf on l.external_language_id = lf.external_language_id
+JOIN sys.database_principals dp on l.principal_id = dp.principal_id
+ORDER BY l.external_language_id, platform";
+
+        private static string GetLanguageScript = $@"
+SELECT l.external_language_id, language, l.create_date, dp.name, content, file_name, platform_desc, parameters, environment_variables
+FROM  sys.external_languages l 
+JOIN sys.external_language_files lf on l.external_language_id = lf.external_language_id
+JOIN sys.database_principals dp on l.principal_id = dp.principal_id
+WHERE language=@{LanguageNameParamName}
+ORDER BY platform";
+
+        public const string CreateScript = "CREATE EXTERNAL LANGUAGE";
+        public const string DropScript = "DROP EXTERNAL LANGUAGE";
+        public const string AlterScript = "ALTER EXTERNAL LANGUAGE";
+        public const string SetContentScript = "SET";
+        public const string AddContentScript = "ADD";
+        public const string RemoveContentScript = "REMOVE";
+        private const string ContentParamName = "CONTENT";
+        private const string FileNameParamName = "FILE_NAME";
+        private const string PlatformParamName = "PLATFORM";
+        private const string EnvVariablesParamName = "ENVIRONMENT_VARIABLES";
+        private const string ParametersParamName = "PARAMETERS";
+        private const string LanguageNameParamName = "LANGUAGE";
+
+        private string GetDropScript(string languageName)
+        {
+            return $@"{DropScript} [{languageName}]";
+        }
+
+        /// <summary>
+        /// Returns the status of external languages in a connection
+        /// </summary>
+        /// <param name="connection"></param>
+        /// <param name="languageName"></param>
+        /// <returns></returns>
+        public virtual bool GetLanguageStatus(IDbConnection connection, string languageName)
+        {
+            bool status = false;
+            try
+            {
+                using (IDbCommand command = connection.CreateCommand())
+                {
+                    command.CommandText = StatusScript;
+                    var parameter = command.CreateParameter();
+                    parameter.ParameterName = $"@{LanguageNameParamName}";
+                    parameter.Value = languageName;
+                    command.Parameters.Add(parameter);
+                    using (IDataReader reader = command.ExecuteReader())
+                    {
+                        while (reader.Read())
+                        {
+                            status = (Convert.ToBoolean(reader[0].ToString()));
+                        }
+                    }
+                }
+            }
+            catch
+            {
+                status = false;
```

Comment:
Not going to log the error at all? Seems like something that should at least go into the debug log.